src/components/NumericStepper/__tests__/NumericStepper.spec.tsx

```diff
@@ -0,0 +1,278 @@
+
+        // Increase by 2 (to equal to 4) and make sure that number of onChange calls equal to 2
+        await increaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(2);
+        expect(onChangeMock).toBeCalledWith(4);
+        expect(inputButton).toHaveAttribute('value', '4');
+        expect(onChangeMock).toHaveBeenCalledWith(4);
+    });
+
+    it('should disable buttons when limits are reached', async () => {
+        const user = userEvent.setup();
+        view = render(
+            <NumericStepper onChange={onChangeMock} minValue={1} maxValue={3} defaultValue={2} />
+        );
+        const inputButton = screen.getByRole('spinbutton');
+        const stepperButtons = screen.getAllByRole('button');
+
+        // Simulate two clicks on stepUp button
+        await increaseClick();
+
+        expect(inputButton).toHaveAttribute('value', '3');
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(stepperButtons[1]).toHaveAttribute('disabled');
+        expect(onChangeMock).toHaveBeenCalledWith(3);
+
+        // Simulate two clicks on stepDown button
+        await user.click(stepperButtons[0]);
+
+        expect(onChangeMock).toBeCalledTimes(2);
+
+        await user.click(stepperButtons[0]);
+
+        expect(onChangeMock).toBeCalledTimes(3);
+        expect(inputButton).toHaveAttribute('value', '1');
+        expect(stepperButtons[0]).toHaveAttribute('disabled');
+    });
+
+    it('should set bottom edge value if user enters value below allowed limit', async () => {
+        const user = userEvent.setup();
+        view = render(<NumericStepper onChange={onChangeMock} minValue={10} />);
+        const inputButton = screen.getByRole('spinbutton');
+
+        await user.type(inputButton, '9');
+
+        expect(inputButton).toHaveAttribute('value', '10');
+    });
+
+    it('should set top edge value if user enters value above allowed limit', async () => {
+        const user = userEvent.setup();
+        view = render(<NumericStepper onChange={onChangeMock} maxValue={10} />);
+        const inputButton = screen.getByRole('spinbutton');
+
+        await user.type(inputButton, '11');
+        await inputButton.blur();
+
+        expect(inputButton).toHaveAttribute('value', '10');
+    });
+
+    it('should set value to the previous one when user enters invalid value', async () => {
+        const user = userEvent.setup();
+        view = render(
+            <NumericStepper onChange={onChangeMock} minValue={2} maxValue={4} defaultValue={3} />
+        );
+
+        await increaseClick();
+
+        expect(onChangeMock).toBeCalledTimes(1);
+        expect(onChangeMock).toBeCalledWith(4);
+        expect(screen.getByRole('spinbutton')).toHaveAttribute('value', '4');
+        expect(onChangeMock).toHaveBeenCalledWith(4);
+
+        const inputButton = screen.getByRole('spinbutton');
+        // Imagine users enters empty string
+        await user.type(inputButton, '   ');
+
+        inputButton.blur();
```

Comment:
should we wait for this? In the previous test you wrote it with `await`

Reply:
i think yes